File: app/src/main/java/org/phenoapps/intercross/IdEntryContract.java
package org.phenoapps.intercross;

import android.provider.BaseColumns;

final class IdEntryContract {

    private IdEntryContract() {}

    static class IdEntry implements BaseColumns {
        static final String TABLE_NAME = "INTERCROSS";
        static final String COLUMN_NAME_ID = "_id";
        static final String COLUMN_NAME_MALE = "male";
        static final String COLUMN_NAME_FEMALE = "female";
        static final String COLUMN_NAME_CROSS = "cross_id";
        static final String COLUMN_NAME_USER = "person";
        static final String COLUMN_NAME_DATE = "timestamp";
        static final String COLUMN_NAME_LOCATION = "location";
    }

    static final String SQL_CREATE_ENTRIES =
            "CREATE TABLE INTERCROSS( " +
                    IdEntry.COLUMN_NAME_ID + " TEXT PRIMARY KEY, "
                    + IdEntry.COLUMN_NAME_MALE + " TEXT,"
                    + IdEntry.COLUMN_NAME_FEMALE + " TEXT,"
                    + IdEntry.COLUMN_NAME_CROSS + " TEXT,"
                    + IdEntry.COLUMN_NAME_USER + " TEXT,"
                    + IdEntry.COLUMN_NAME_DATE + " TEXT,"
                    + IdEntry.COLUMN_NAME_LOCATION + " TEXT)";

    static final String SQL_DELETE_ENTRIES =
            "DROP TABLE IF EXISTS " + IdEntry.TABLE_NAME;

}

File: app/src/main/java/org/phenoapps/intercross/IdEntryContract.kt
package org.phenoapps.intercross

import android.provider.BaseColumns

internal object IdEntryContract {


    const val TABLE_NAME = "INTERCROSS"

    const val SQL_CREATE_ENTRIES =
            "CREATE TABLE $TABLE_NAME(${IdEntry.COLUMN_NAME_ID} INTEGER PRIMARY KEY," +
                    "${IdEntry.COLUMN_NAME_CROSS} TEXT, ${IdEntry.COLUMN_NAME_FEMALE}," +
                    "${IdEntry.COLUMN_NAME_MALE} TEXT, ${IdEntry.COLUMN_NAME_USER}," +
                    "${IdEntry.COLUMN_NAME_DATE} TEXT, ${IdEntry.COLUMN_NAME_LOCATION} TEXT," +
                    "${IdEntry.COLUMN_NAME_POLLINATION_TYPE} TEXT, ${IdEntry.COLUMN_NAME_CROSS_COUNT} INTEGER," +
                    "${IdEntry.COLUMN_NAME_CROSS_NAME} TEXT)"

    val SQL_DELETE_ENTRIES = "DROP TABLE IF EXISTS " + IdEntry.TABLE_NAME

    internal class IdEntry : BaseColumns {
        companion object {
            const val TABLE_NAME = "INTERCROSS"
            const val COLUMN_NAME_ID = "_id"
            const val COLUMN_NAME_MALE = "male"
            const val COLUMN_NAME_FEMALE = "female"
            const val COLUMN_NAME_CROSS = "cross_id"
            const val COLUMN_NAME_USER = "person"
            const val COLUMN_NAME_DATE = "timestamp"
            const val COLUMN_NAME_LOCATION = "location"
            const val COLUMN_NAME_POLLINATION_TYPE = "cross_type"
            const val COLUMN_NAME_CROSS_COUNT = "cross_count"
            const val COLUMN_NAME_CROSS_NAME = "cross_name"

            val COLUMNS = arrayOf(COLUMN_NAME_ID, COLUMN_NAME_MALE, COLUMN_NAME_FEMALE,
                    COLUMN_NAME_CROSS, COLUMN_NAME_USER, COLUMN_NAME_DATE, COLUMN_NAME_LOCATION,
                    COLUMN_NAME_POLLINATION_TYPE, COLUMN_NAME_CROSS_COUNT, COLUMN_NAME_CROSS_NAME)
        }
    }

}
